python/pyspark/__init__.py

```diff
@@ -36,6 +36,31 @@
       Finer-grained cache persistence levels.
 
 """
+import os
+import sys
+
+import xml.etree.ElementTree as ET
+
+if (os.environ.get("SPARK_HOME", "not found") == "not found"):
```

Comment:
I don't really understand this part of the code, so it would be nice to get some core devs to chime in, but it looks like

    ./core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

does a lot of logic, especially when deploying against YARN that seems important.